docs/src/user-guide/reference-json-search-syntax.md

```diff
@@ -0,0 +1,205 @@
+wildcard expression, where the wildcard `*` which matches zero or more characters:
+
+```
+key: "*partial value*"
+```
+
+:::{caution}
+Wildcard expressions only work for kv-pairs where the value is a string. 
+:::
+
+:::{caution}
+CLP doesn't currently support the `?` wildcard (which matches any single character) _except_ in
+values containing multiple words. This limitation will be addressed in a future version of CLP. 
+:::
+
+### Wildcard queries for keys
+
+To search for a kv-pair with *any* key, you can specify the query in one of two ways:
+
+```
+value
+```
+
+OR
+
+```
+*: value
+```
+
+To search for a kv-pair where only some parts of a nested key are known, you can replace the unknown
+parts with the `*` wildcard:
+
+```
+parent1.*.parent3.child: value
+```
+
+:::{note}
+CLP does not support queries for partial keys like `parent1*`.
+:::
+
+### Numeric comparisons
+
+To search for a kv-pair where the value is a _number_ in some range, you can use numeric comparison
+operators in place of the `:`. For example:
+
+```
+key > value
+```
+
+The following comparison operators are supported:
+
+* `>` - the kv-pair's value is greater than the specified value
+* `>=` - the kv-pair's value is greater than or equal to the specified value
+* `<` - the kv-pair's value is less than the specified value
+* `<=` - the kv-pair's value is less than or equal to the specified value
+
+:::{note}
+There is no `=` operator since `:` functions as an equality operator.
+:::
+
+### Querying array values
+
+To search for a kv-pair where the value is in an array, the syntax is the same as searching for a
+nested kv-pair. For example, the query below...
+
+```
+parent1: {parent2: {child: value}}
+```
+
+...would match the log event below:
+
+```json
+{"parent1": [{"parent2": {"child": "value"}}]}
+```
+
+:::{caution}
+CLP does not currently support queries for array kv-pairs where only part of the key is known. In
+other words, the key must either be a wildcard (`*`) or it must contain no wildcards.
+:::
+
+### Complex queries
+
+You can search for one or more kv-pairs by combining them with boolean algebra. For example:
+
+```text
+key1: value1 AND (key2: valueA OR key2: valueB) AND NOT key3: value3
+```
+
+There are three supported boolean operators:
+
+* `AND` - the expressions on _both_ sides of the operator must be true. 
+* `OR` - the expressions on _either_ side of the operator must be true.
+* `NOT` - the expression after the operator must _not_ be true.
+
+You can use parentheses (`()`) to apply an operator to a group of expressions.
+
+## Examples
+
+**Search for log events that contain a specific key-value pair:**
+
+```
+id: 22149
+```
+
+**Search for ERROR log events containing a substring:**
+
+```
+level: ERROR AND message: "*job*"
+```
+
+**Search for FATAL or ERROR log events:**
+
+```
+level: FATAL OR level: ERROR
+```
+
+(differences-with-kql)=
+## Differences with KQL
```

Comment:
I think there are two other differences. 1) In clp, If a user wants to search something as a substring of a value, they need to explicitly add `*`. 2) clp has `date` comparison support.

Reply:
Do you mean that if the user enters a query like `key: value`, in KQL, that will still be interpreted as `key: *value*`?

Reply:
I haven't tried it, but yeah I think so since in the document query `http.request.body.content: null pointer` is to search for documents where `http.request.body.content` (a text field) contains the text "null pointer"

Reply:
For 2), how confident are we that we can support it long term and/or do we want to improve the implementation of timestamp parsing before we mention it? What do you think, @gibber9809?

Reply:
> I haven't tried it, but yeah I think so since in their docs query `http.request.body.content: null pointer` is to search for documents where `http.request.body.content` (a text field) contains the text "null pointer"

Gotcha, I think that's a fuzzy match case in KQL. I'll mention it.

Reply:
Just a note on (1) the semantic we have right now is that if a user adds any explicit wildcards we don't add any implicit ones, and we only add implicit wildcards at the start and end if the predicate contains a space.

For (2) let's not mention it for now. We only allow that date syntax on the main timestamp column specified at ingestion time, so users will probably get confused when they try it on some other column containing a date and it doesn't work.

Reply:
Hm, I think that semantic is a little unintuitive. A user would have to remember:

* If the query value contains a single word, then it's an exact match unless the value contains wildcards.
* If the query value contains a space, then it's a substring match...
  * ...unless there are wildcards, in which case it's still a substring match...
    * ...unless  the query has no prefix or suffix wildcard, in which case the query is anchored to the beginning/end of the value.

It might be clearer for the user if we don't add implicit wildcards. They'd have to get used to inserting wildcards themselves, but it would give them the most flexibility and predictability.

Reply:
Yeah, we can make that change.